Add a chapter about the new pipeline syntax to the 3.0 blog post (#2921)
tenzir/vast: web/blog/vast-v3.0/index.md

[kusto]: https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/

The new syntax resembles the well-known Unix paradigm of command chaining. The
difference to Unix pipelines is that VAST exchanges structured data between
operators. The `vast export` and `vast import` commands accept such a
pipeline as a string argument. Previously, VAST only supported a YAML-like
definition of pipeline in configuration files. This new, ad-hoc form enables
much quicker data transformation.

Refer to the [pipelines documentation][pipeline-doc] for more details on how to
use the new pipeline syntax.

:::info Pipeline YAML Syntax Deprecation
This release introduces a transitional period from YAML-style to Unix-style
pipelines. The old YAML syntax for pipelines will be deprecated and removed
altogether in a future version. The new pipeline operators [`head`][head-op] and
[`taste`][taste-op] have no YAML equivalent.
:::

[pipeline-doc]: /docs/understand/query-language/pipelines#define-a-pipeline
[head-op]: /docs/understand/query-language/operators/head
[taste-op]: /docs/understand/query-language/operators/taste

## VASTQL Upgrades

> We've done some breaking changes to the the **VAST** **Q**uery **L**anguage

Author: https://github.com/tenzir/vast/pull/2921/commits/84fd4d246c79883628c95a6410e3dd0bd13bd5a0

> that we've wanted to do in quite a while. Here's a summary:

Reviewer: ```suggestion
that we've wanted to do for a long time. Here's a summary:
```

Author: https://github.com/tenzir/vast/pull/2921/commits/84fd4d246c79883628c95a6410e3dd0bd13bd5a0

<!--truncate-->

> ## Unix-Style Pipelines
> 
> Starting from v3.0, VAST introduces new and more dynamic ways to execute
> pipelines. Aside from using the YAML syntax for pipelines in a configuration
> file, it is now possible to define a Unix-style chain of pipeline operators in
> a single string. The `vast export` and `vast import` commands accept such a
> pipeline as a string parameter. The commands will then instantly evaluate and
> execute the pipeline. This new functionality allows for quicker and more
> responsive data transformation.
> 
> Refer to the ["Pipelines" section of the documentation][pipeline-doc] for more
> details on how to use the new Unix-style pipeline syntax.
> 
> :::info Pipeline YAML Syntax Deprecation
> This change introduces a transitional period from YAML-style to Unix-style
> pipelines. The old YAML syntax for pipelines and for pre-v3.0 pipeline operators
> will be deprecated and removed altogether in a future version. The new pipeline
> operators [`head`][head-op] and [`taste`][taste-op] do not already contain a
> YAML syntax. The syntax of some pre-v3.0 pipeline operators will also be
> subject to change/deprecation.
> :::

Reviewer: ```suggestion
## Unix-Style Pipelines

Starting with v3.0, VAST introduces a new way to write pipelines, with a syntax
similar to [splunk](https://splunk.com), [Kusto][kusto],
[PRQL](https://prql-lang.org/), and [Zed](https://zed.brimdata.io/).

[kusto]: https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/

The new syntax resembles the well-known Unix paradigm of command chaining. The
difference to Unix pipelines is that VAST exchanges structured data between
operators. The `vast export` and `vast import` commands accept such a
pipeline as a string argument. Previously, VAST only supported a YAML-like
definition of pipeline in configuration files. This new, ad-hoc form enables
much quicker data transformation.

Refer to the [pipelines documentation][pipeline-doc] for more details on how to
use the new pipeline syntax.

:::info Pipeline YAML Syntax Deprecation
This release introduces a transitional period from YAML-style to Unix-style
pipelines. The old YAML syntax for pipelines will be deprecated and removed
altogether in a future version. The new pipeline operators [`head`][head-op] and
[`taste`][taste-op] have no YAML equivalent.
:::
```

> [pipeline-doc]: //vast.io/docs/understand/query-language/pipelines#define-a-pipeline
> [head-op]: //vast.io/docs/understand/query-language/operators/head
> [taste-op]: //vast.io/docs/understand/query-language/operators/taste

Reviewer: ```suggestion
[pipeline-doc]: /docs/understand/query-language/pipelines#define-a-pipeline
[head-op]: /docs/understand/query-language/operators/head
[taste-op]: /docs/understand/query-language/operators/taste
```

## VASTQL Upgrades

> The 3.x series of VAST releases is also all about the **VAST** **Q**uery

Reviewer: The intro now longer makes sense. This comes *after* the introduction of Unix-style pipelines.

Author: I shortened the intro paragraph in https://github.com/tenzir/vast/pull/2921/commits/d19819131aab6ec509ef5d860e7c7346c1e56b3c

> **L**anguage. To kick this off, we've done some breaking changes to the language

Reviewer: "To kick this off"

👆 this used to be the intro paragraph where this phrasing made sense, but needs to be reworded now.

Author: I shortened the intro paragraph in https://github.com/tenzir/vast/pull/2921/commits/d19819131aab6ec509ef5d860e7c7346c1e56b3c

## Unix-Style Pipelines

Starting with v3.0, VAST introduces a new way to write pipelines, with a syntax
similar to [splunk](https://splunk.com), [Kusto][kusto],
[PRQL](https://prql-lang.org/), and [Zed](https://zed.brimdata.io/).

[kusto]: https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/

The new syntax resembles the well-known Unix paradigm of command chaining. The
difference to Unix pipelines is that VAST exchanges structured data between
operators. The `vast export` and `vast import` commands accept such a
pipeline as a string argument. Previously, VAST only supported a YAML-like

> definition of pipeline in configuration files. This new, ad-hoc form enables

Reviewer: ```suggestion
definition of pipelines in configuration files. This new, ad-hoc form enables
```
